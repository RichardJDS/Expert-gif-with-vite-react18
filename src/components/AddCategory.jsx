import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState('');
    const handleInputChange=(e)=>{

        setInputValue(e.target.value);
    }
    
    const handleSubmit = (e)=>{
        e.preventDefault();
        if (inputValue.trim().length > 2){
            onNewCategory(inputValue);
        // setCategories(c=>{
        //     if(!c.includes(inputValue)){
        //         return [inputValue, ...c];
        //     }
        //      else{
        //     return c}});
        setInputValue('');
        }
    }
    
    console.log("Se ejecuto el teclado")
  return (
    <form onSubmit={handleSubmit} aria-label ="form">
    <h2>{inputValue}</h2>
    <input 
    type="text"
    value={inputValue}
    onChange={handleInputChange}
    />
    </form>
  )
}
AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}