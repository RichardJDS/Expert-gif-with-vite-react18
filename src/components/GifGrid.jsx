import React, { useEffect, useState } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './GifItem';

// console.log("Se ejecuto fuera de Escope gifgrid"); SOLO SE EJECUTA UNA VEZ
// POR MAs que se vuelva a renderizar nuevamente o se envie un nuevo props

export const GifGrid = ({category}) => {
  console.log("Se ejecuto gifgrid")
  
  const {images, loading} = useFetchGifs(category);

  const [count, setCount] = useState(0);
  // Cuando se realiza la ejecucion de una funcion PARA CCAMBIAR EL ESTADO DEL HOOK
  // SOLO se ejecute el componente donde se encuentra ese HOOK, no afecta AL COMPONENTE
  // PADRE NO AL COMPONENTE PADRE, puede ejecutar otro compoente si tiene adentro otro componente
  // pero el hijo no ejecuta al padre


  return (
    <div>
        <button onClick={()=>setCount()}>+1</button>
        {loading && <p>Cargando.....</p>}
        <h3>{category}</h3>
        <ol>
          {
            images.map((img)=>{
            return (<GifItem key={img.id} img = {img}/>);
            } )
          }
        </ol>

    </div>
  )
}
