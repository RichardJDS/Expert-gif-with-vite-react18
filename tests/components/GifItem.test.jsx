import { render, screen } from "@testing-library/react";
import { scryRenderedComponentsWithType } from "react-dom/test-utils";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en el componente GifItem,', () => { 
    const img = {
        title:"Paisajes",
        url: "https://paisajes.jpg/"
    }
    test('Debe de hacer Match con el SnapShot', () => { 
        const {container} = render(<GifItem img={img}/>);
        expect(container).toMatchSnapshot();
    });
    test('Debe de mostrar la imagen con la url y el alt indicado', () => { 
        render(<GifItem img={img} />);
        const {src, alt} = screen.getByRole("img");
        screen.debug();
        expect(src).toBe(img.url);
        expect(alt).toBe(img.title);
     });
     test('should to show the title in the image', () => { 
        render(<GifItem img={img} />);
        expect(screen.getByText(img.title)).toBeTruthy();
      });
 });