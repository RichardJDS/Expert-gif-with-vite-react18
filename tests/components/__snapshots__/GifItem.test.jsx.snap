// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifItem, Debe de hacer Match con el SnapShot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Paisajes"
      src="https://paisajes.jpg/"
    />
    <p>
      Paisajes
    </p>
  </div>
</div>
`;
